README.md

```diff
@@ -1,52 +1,105 @@
+global `ssh_config` that matches OS default (the generated configuration does
+not keep comments and order of the options).
 
-Variables that are not intended as input, like variables defined in
-`vars/main.yml`, variables that are read from other roles and/or the global
-scope (ie. hostvars, group vars, etc.) can be also mentioned here but keep in
-mind that as these are probably not part of the role API they may change during
-the lifetime.
+ * `ssh_user`:
 
-Example of setting the variables:
+By default (*null*) the role will modify the global configuration for all
+users. Other values will be interpretted as a username and the role will
+modify per-user configuration stored under `~/.ssh/config` of the given user.
+The user needs to exists before invoking this role otherwise it will fail.
 
-```yaml
-ssh_foo: "oof"
-ssh_bar: "baz"
-```
+ * `ssh_skip_defaults`:
+
+By default (`auto`), the role writes the system-wide configuration file
+`/etc/ssh/ssh_config` and keeps OS defaults defined there (*true*). This is
+automatically disabled, when a drop-in configuration file is created
+(`ssh_drop_in_name!=null`) or when per-user configuration file is created
+(`ssh_user!=null`).
+
+ * `ssh_drop_in_name`:
+
+This defines the name for the drop-in configuration file to be placed in
+system-wide drop-in directory. The name is used in the template
+defined by `__ssh_drop_in_template` (by default
+`/etc/ssh/ssh_config.d/{name}.conf`) to reference the configuration file
+to be modified. If the system does not support drop-in directory, setting
+this option will make the play fail. Default is `null` if the system does
+not support drop in directory and `00-ansible` otherwise.
+
+The suggested format is `NN-name`, where `NN` is two-digit number used for
+sorting the and `name` is any descriptive name for the content or the owner
+of the file.
+
+ * `ssh`:
+
+A dict containing configuration options and respective values. See example
+below.
+
+ * `ssh_...`:
+
+Simple variables consisting of the option name prefixed with `ssh_` can be
+used rather than a dict above. The simple variable overrides values in dict
+above.
 
-### Variables Exported by the Role
+ * `ssh_additional_packages`:
 
-This section is optional.  Some roles may export variables for playbooks to
-use later.  These are analogous to "return values" in Ansible modules.  For
-example, if a role performs some action that will require a system reboot, but
-the user wants to defer the reboot, the role might set a variable like
-`ssh_reboot_needed: true` that the playbook can use to reboot at a more
-convenient time.
+This role automatically installs packages needed for most common use cases
+on given platform. If some additional packages need to be installed (for
+example `openssh-keysign` for hostbased authentication), they can be specified
+in this variable.
 
-Example:
+### Secondary role variables:
+
+These variables are used by the role internals and can be used to override
```

Comment:
It means the secondary role variables are not to be set by the users?

It's a bit confusing to me since, e.g.,  __ssh_packages is declared in vars/main.yml, which cannot be reset; while ssh_config_file is in defaults/main.yml, which is allowed to set (as being tested in tests_custom_config.yml and tests_all_options.yml.)

Reply:
It was still quite a mix as I was not quite sure where to draw a line. The underscored variables should be only internal and should be probably mentioned in the `contributing.md`. I am moving them in there. The rest in this section should probably come up to normal variables as they can be still used.

Using `__ssh_drop_in_template` is not encouraged as it should not be needed in normal cases, but it might be useful for generating drop-in snippets in different location (not expecting this would be common use case).

Reply:
ok - if it is part of the public API, we should not change it

Reply:
Moved to internal `contributing.md` and `README.md` now mentions only the default path, not template variable name.